tests/transforms/test_split_non_commuting.py

```diff
@@ -122,765 +119,580 @@ def complex_wires_processing_fn(results):
     )
 
 
-# Measurements that accept observables as arguments
-obs_measurements = [qml.expval, qml.var, qml.probs, qml.counts, qml.sample]
-
-# measurements that accept wires as arguments
-wire_measurements = [qml.probs, qml.counts, qml.sample]
-
+# contains the following observables: X(0), Y(0), Y(0) @ Z(1), X(1), H(0) @ X(1), Projector(1)
+# wires groups: ((0, 3), (1, 5), (2,), (4,))
+non_pauli_obs_list = [
+    qml.X(0),
+    0.5 * qml.Y(0),
+    qml.X(0) + qml.Y(0) @ qml.Z(1) + 2.0 * qml.X(1) + qml.H(0) @ qml.PauliX(wires=[1]),
+    qml.Projector([0, 1], wires=1),
+]
 
-class TestUnits:
-    """Unit tests for components of the ``split_non_commuting`` transform"""
+non_pauli_obs_wires_groups = [
+    [qml.X(0), qml.X(1)],
+    [qml.Y(0), qml.Projector([0, 1], wires=1)],
+    [qml.Y(0) @ qml.Z(1)],
+    [qml.H(0) @ qml.X(1)],
+]
 
-    @pytest.mark.parametrize("measure_fn", obs_measurements)
-    @pytest.mark.parametrize(
-        "grouping_strategy, n_tapes", [(None, 5), ("default", 2), ("qwc", 2), ("wires", 4)]
-    )
-    def test_number_of_tapes(self, measure_fn, grouping_strategy, n_tapes):
-        """Tests that the correct number of tapes is returned"""
 
-        measurements = [measure_fn(op=o) for o in single_term_obs_list]
-        tape = qml.tape.QuantumScript([qml.X(0), qml.CNOT([0, 1])], measurements, shots=100)
-        tapes, _ = split_non_commuting(tape, grouping_strategy=grouping_strategy)
-        assert len(tapes) == n_tapes
-        assert all(t.operations == [qml.X(0), qml.CNOT([0, 1])] for t in tapes)
-        assert all(t.shots == tape.shots for t in tapes)
+def non_pauli_obs_processing_fn(results):
+    """The expected processing function for non-Pauli observables"""
 
-    @pytest.mark.parametrize(
-        "grouping_strategy, n_tapes", [(None, 5), ("default", 2), ("qwc", 2), ("wires", 4)]
-    )
-    @pytest.mark.parametrize(
-        "make_H",
-        [
-            lambda coeffs, obs: qml.Hamiltonian(coeffs, obs),
-            lambda coeffs, obs: qml.sum(*(qml.s_prod(c, o) for c, o in zip(coeffs, obs))),
-        ],
-    )
-    def test_number_of_tapes_single_hamiltonian(self, grouping_strategy, n_tapes, make_H):
-        """Tests that the correct number of tapes is returned for a single Hamiltonian"""
+    group0, group1, group2, group3 = results
+    return group0[0], 0.5 * group1[0], group0[0] + group2 + 2.0 * group0[1] + group3, group1[1]
```

Comment:
```suggestion
    res = (group0[0], 0.5 * group1[0], group0[0] + group2 + 2.0 * group0[1] + group3, group1[1])
    return res
```

Could we do something like this here and above? It makes step-wise debugging easier to inspect.